### PR TITLE
Nornir Filtering Deep Dive How To

### PR DESCRIPTION
**Pull request by writememe**
Hi nornir maintainers,

As discussed with @dbarrosop in #643, I've created a filtering deep dive how to, to add to the official docs. 

It's pretty detailed, but hopefully you are happy with it and it makes sense.

### CODE REVIEW

**Comment by writememe** [comment]
## Jupyter test passing on my local machine using python3.8:

```
➜  nornir git:(develop) make nbval
poetry run pytest --nbval --sanitize-with docs/nbval_sanitize.cfg \
        docs/tutorial/ \
        docs/howto/
================================================= test session starts ==================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /mnt/c/Users/dfjt/Documents/networking/projects/nornir-docs-dev/nornir, configfile: setup.cfg
plugins: nbval-0.9.6, pylama-7.7.1, cov-2.11.1, requests-mock-1.8.0
collected 117 items

docs/tutorial/failed_tasks.ipynb ................                                                                [ 13%]
docs/tutorial/initializing_nornir.ipynb ......                                                                   [ 18%]
docs/tutorial/inventory.ipynb .................................                                                  [ 47%]
docs/tutorial/processors.ipynb ......                                                                            [ 52%]
docs/tutorial/task_results.ipynb .............                                                                   [ 63%]
docs/tutorial/tasks.ipynb ........                                                                               [ 70%]
docs/howto/advanced_filtering.ipynb ...........                                                                  [ 79%]
docs/howto/filtering_deep_dive.ipynb ....................                                                        [ 96%]
docs/howto/handling_connections.ipynb ....                                                                       [100%]

=================================================== warnings summary ===================================================
../../../../../../../../../home/dfjt/.cache/pypoetry/virtualenvs/nornir-2dBYvXJ5-py3.8/lib/python3.8/site-packages/_pytest/config/__init__.py:1233
  /home/dfjt/.cache/pypoetry/virtualenvs/nornir-2dBYvXJ5-py3.8/lib/python3.8/site-packages/_pytest/config/__init__.py:1233: PytestConfigWarning: Unknown config option: python_paths

    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================================== 117 passed, 1 warning in 23.69s ============================================
➜  nornir git:(develop)
```

**Comment by writememe** [comment]
Tests are now passing. There were issues with my yaml files. Now ready for review.

File: docs/howto/filtering_deep_dive.ipynb [hunk 1]
```diff
@@ -0,0 +1,2195 @@
...
+    "print(f\"Region is: {single_group_region}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "creative-hearts",
+   "metadata": {},
+   "source": [
+    "By now, you should have an understanding of the following regarding inventory custom data:  \n",
+    "    - __What is nornir custom data?__  \n",
```
Comment (dbarrosop) [inline docs/howto/filtering_deep_dive.ipynb:911]: check how this is being formatted, not sure if it was your intention to render the underscores or if you were looking for italic or something else

**Comment by dbarrosop** [comment]
Looks really good, just a couple of comments:

1. Do you want to publish a pointer a blog post in `https://nornir.tech/posts/` to announce we are publishing this?
2. It is quite extensive, maybe it's worth turning it into a "series" and split it into different documents? I personally don't mind long documents but looks like nowadays everybody has some form of ADHD :squirrel:

**Comment by dbarrosop** [comment]
On a second look, it is not as long, there is a lot of output and code examples and looks long but it isn't.

**Comment by writememe** [comment]
> 
> 
> On a second look, it is not as long, there is a lot of output and code examples and looks long but it isn't.

Great, glad you agree. A lot of it is quite verbose and a lot of just printing outputs. I thought it would be better to give more detail and hopefully it's received well.

File: docs/howto/filtering_deep_dive.ipynb (hunk 1 above)
Comment (writememe) [reply docs/howto/filtering_deep_dive.ipynb]: Thanks for picking this up. Yep, I meant for this to be italic and have now resolved it.

**Comment by writememe** [comment]
> 
> 
> Looks really good, just a couple of comments:
> 
>     1. Do you want to publish a pointer a blog post in `https://nornir.tech/posts/` to announce we are publishing this?
> 
For sure @dbarrosop, is this where you need me to create a markdown post into (obviously in a new 2021 folder):

https://github.com/nornir-automation/nornir.tech.src/tree/master/content/posts/

I can write something up, and we can announce it from there. I feel like this is a good way of communicating from the official project.

**Comment by writememe** [comment]
I think it's best to merge this code first and make sure all that works. But I've done all the work for the blog post and the PR is ready for the website here:

https://github.com/nornir-automation/nornir.tech.src/pull/23

**Comment by dbarrosop** [comment]
Sounds good to me, merging!

**Pull request merged**